xarray/core/computation.py

```diff
@@ -1950,14 +1950,15 @@ def _ensure_numeric(data: Dataset | DataArray) -> Dataset | DataArray:
     from .dataset import Dataset
 
     def to_floatable(x: DataArray) -> DataArray:
-        if x.dtype.kind == "M":
-            # datetimes
+        if x.dtype.kind in "MO":
+            # datetimes (CFIndexes are object type)
+            offset = (
+                np.datetime64("1970-01-01")
+                if x.dtype.kind == "M"
+                else type(x.data[0])(1970, 1, 1)
```

Comment:
```suggestion
                else type(x.data[0].compute())(1970, 1, 1)
```

The offset needs to be of the right type, so I don't think we can avoid it. You'll need to pass `max_computes=1` to `raise_if_dask_computes` for the tests to pass.

Reply:
Probably also need #6556